app/which_shop.py

```diff
@@ -0,0 +1,96 @@
+from app.distance import Distance
+import datetime
+
+
+class Shop:
+
+    def __init__(
+            self,
+            customer: dict,
+            shop: dict,
+            dct: dict
+    ):
+        self._customer = customer

```

Comment:
customer shouldn't be an attribute of Shop instance. Customers come and go. They are not tied to shop